src/files.js

```diff
@@ -61,14 +61,17 @@ function doUpload (cozy, data, method, path, options) {
     lastModifiedDate = new Date(lastModifiedDate)
   }
 
+  const headers = {
+    'Content-Type': contentType
+  }
+  if (contentLength) headers['Content-Length'] = String(contentLength)
+  if (checksum) headers['Content-MD5'] = checksum
+  if (lastModifiedDate) headers['Date'] = lastModifiedDate.toGMTString()
+  if (ifMatch) headers['If-Match'] = ifMatch
```

Comment:
Have you noticed a problem when a header was set to '' ?

Reply:
I had an issue with an undefined header wich was sent as "undefined", I feel this better express the need.

Reply:
👍